### PR TITLE
Update bytebuddy, add tab completion for PacketLogging command and use cache for getFieldAccessor

### PR DESCRIPTION
**Pull request by camdenorrb**
- ByteBuddy seemed to be out of date so I updated it
- Packet logging command seemed a bit complex, so I added tab completion to make it easier to use
- `getFieldAccessor` seemed a bit slow on our production servers so I went ahead and added a timed cache for it.

While getFieldAccessor was using a cache, it seemed that the gc was clearing it frequently

### CODE REVIEW

**Review by derklaro — COMMENTED** [review]
I think the packet logging tab complete *might* be reasonable, but the field accessor cache is not. It would be much better to find the places where field accessors are requested and update the code to use fields for them, rather than caching them in a map.

**Comment by camdenorrb** [comment]
@derklaro it was just a quick fix while keeping to pattern

File: src/main/resources/plugin.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: ProtocolLib
 version: ${project.fullVersion}
 description: Provides read/write access to the Minecraft protocol.
-authors: [dmulloy2, comphenix]
+authors: [dmulloy2, comphenix, camdenorrb]
```
Comment (dmulloy2) [inline src/main/resources/plugin.yml:4]: no

File: src/main/java/com/comphenix/protocol/PacketLogging.java [hunk 2]
```diff
@@ -163,6 +162,31 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
 		}
 	}
 
+	@Nullable
+	@Override
+	public List<String> onTabComplete(@NotNull final CommandSender commandSender, @NotNull final Command command, @NotNull final String s, @NotNull final String[] strings) {
```
Comment (dmulloy2) [inline src/main/java/com/comphenix/protocol/PacketLogging.java:167]: looks good

File: src/main/java/com/comphenix/protocol/collections/ExpireHashMap.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public String toString() {
 	}
 	
 	private final Map<K, ExpireEntry> keyLookup = new HashMap<>();
-	private final PriorityQueue<ExpireEntry> expireQueue =  new PriorityQueue<>();
+	private final PriorityBlockingQueue<ExpireEntry> expireQueue =  new PriorityBlockingQueue<>();
```
Comment (dmulloy2) [inline src/main/java/com/comphenix/protocol/collections/ExpireHashMap.java:52]: reason behind this change?

File: src/main/java/com/comphenix/protocol/reflect/accessors/MethodHandleHelper.java [hunk 4]
```diff
@@ -75,26 +87,35 @@ public static ConstructorAccessor getConstructorAccessor(Constructor<?> construc
...
-			// java hates us - unreflecting a trusted field always results in an exception, finding them doesn't...
-			MethodHandle getter;
-			MethodHandle setter;
-			if (staticField) {
-				getter = LOOKUP.findStaticGetter(field.getDeclaringClass(), field.getName(), field.getType());
-				setter = LOOKUP.findStaticSetter(field.getDeclaringClass(), field.getName(), field.getType());
-			} else {
-				getter = LOOKUP.findGetter(field.getDeclaringClass(), field.getName(), field.getType());
-				setter = LOOKUP.findSetter(field.getDeclaringClass(), field.getName(), field.getType());
+
```
Comment (dmulloy2) [inline src/main/java/com/comphenix/protocol/reflect/accessors/MethodHandleHelper.java:90]: would prefer if the caching stuff was in a separate PR

File: pom.xml [hunk 5]
```diff
@@ -336,7 +336,7 @@
 		<dependency>
 			<groupId>net.bytebuddy</groupId>
 			<artifactId>byte-buddy</artifactId>
-			<version>1.12.12</version>
+			<version>1.12.18</version>
```
Comment (dmulloy2) [inline pom.xml:339]: any known changes caused by this?

**Review by dmulloy2 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**